app/models/rubygem.rb

```diff
@@ -299,6 +299,8 @@ def ensure_name_format
       errors.add :name, "must include at least one letter"
     elsif name !~ NAME_PATTERN
       errors.add :name, "can only include letters, numbers, dashes, and underscores"
+    elsif name =~ /^[#{Regexp.escape(Patterns::SPECIAL_CHARACTERS)}]+/
```

Comment:
Should `\A` be used instead of `^`here?

Reply:
It probably should, for the sake of correctness, but given newlines aren't valid in `NAME_PATTERN` it isn't strictly necessary